### PR TITLE
cyrus-sasl: Remove private libs from .la files (fixes spice build)

### PR DESCRIPTION
**Pull request by hedning**
###### Motivation for this change

Version 2.1.27 added the private libs to a bunch of .la files, which in turn
broke spice's build, https://github.com/NixOS/nixpkgs/pull/51444#issuecomment-444499394

A cleaner approach would be to just remove the .la files, but I'm unsure if
that will break anything else.

cc #51444 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by hedning** [comment]
@GrahamcOfBorg build spice

**Comment by hedning** [comment]
@GrahamcOfBorg build cyrus_sasl

File: pkgs/development/libraries/cyrus-sasl/default.nix [hunk 1]
```diff
@@ -44,6 +44,7 @@ stdenv.mkDerivation rec {
   postInstall = ''
     for f in $out/lib/*.la $out/lib/sasl2/*.la; do
       substituteInPlace $f --replace "${openssl.dev}/lib" "${openssl.out}/lib"
+      sed -i $f -e 's,-lresolv\|-lgssapi_krb5\|-lkrb5\|-lk5crypto\|-lcom_err,,g'
```
Comment (Mic92) [inline pkgs/development/libraries/cyrus-sasl/default.nix]: Do you basically remove all libraries from the libtool file?

File: pkgs/development/libraries/cyrus-sasl/default.nix [hunk 2]
```diff
@@ -44,6 +39,7 @@ stdenv.mkDerivation rec {
...
```
[the remark this answers is not in the document]
Comment (hedning) [reply pkgs/development/libraries/cyrus-sasl/default.nix]: They're all added to the .la files, without the necessary -L flags.

[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/development/libraries/cyrus-sasl/default.nix]: It comes from here:

https://github.com/cyrusimap/cyrus-sasl/blob/master/m4/sasl2.m4#L121

I would say they are two options: Either put `kerberos` in `propagatedBuildInputs` or disable `kerberos` for cyrus-sasl (maybe enabling via flags would work too).

[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/development/libraries/cyrus-sasl/default.nix]: I am surprised though that you can just drop the libraries without having any missing symbols during linking.

[the remark this answers is not in the document]
Comment (hedning) [reply pkgs/development/libraries/cyrus-sasl/default.nix]: I'm fine with either alternative, though I'm not using this derivation myself, so  cc @eburimu who did the upgrade to 2.1.27. 

I guess downgrading is also an option, the release is very fresh while the previous release is many years old.

[the remark this answers is not in the document]
Comment (pbogdan) [reply pkgs/development/libraries/cyrus-sasl/default.nix]: Just wanted to mention that the commit that added `pruneLibtoolFiles` setup hook has some very useful notes on how to handle `.la` files - https://github.com/NixOS/nixpkgs/commit/fd97db43bcb05e37f6bb77f363f1e1e239d9de53

[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/development/libraries/cyrus-sasl/default.nix]: Ok. So adding `pruneLibtoolFiles` should be safe in our case.

**Comment by hedning** [comment]
Switched to `pruneLibtoolFiles`. Opted to keep the existing openssl modifications in `postInstall`, though it's most likely unnecessary.

**Comment by Mic92** [comment]
It is unnecessary, I checked it.

**Pull request merged**